mesonbuild/interpreter.py

```diff
@@ -3657,8 +3657,17 @@ def func_configure_file(self, node, args, kwargs):
         # Install file if requested, we check for the empty string
         # for backwards compatibility. That was the behaviour before
         # 0.45.0 so preserve it.
-        idir = kwargs.get('install_dir', None)
-        if isinstance(idir, str) and idir:
+        idir = kwargs.get('install_dir', '')
+        if not isinstance(idir, str):
+            raise InterpreterException('"install_dir" must be a string')
+        install = kwargs.get('install', idir != '')
+        if not isinstance(install, bool):
+            raise InterpreterException('"install" must be a boolean')
+        if install:
+            if not idir:
+                raise InterpreterException('"install_dir" must be specified '
+                                           'when "install" in a configure_file '
+                                           'is true')
```

Comment:
Sorry for nitpicking again, but since `install` kwarg was not working before, I think we should declare it as new kwarg:
`@FeatureNewKwargs('configure_file', '0.49.0', ['install'])`